specs/fjord/predeploys.md

```diff
@@ -0,0 +1,39 @@
+# Predeploys
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**
+
+- [GasPriceOracle](#gaspriceoracle)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## GasPriceOracle
+
+Following the Fjord upgrade, three additional values used for L1 fee computation are:
+
+- costIntercept
+- costFastlzCoef
+- costTxSizeCoef
+
+These values are hard-coded constants in the `GasPriceOracle` contract. The
+calculation follows the same formula outlined in the
+[Fjord L1-Cost fee changes (FastLZ estimator)](./exec-engine.md#fjord-l1-cost-fee-changes-fastlz-estimator)
+section.
+
+A new method is introduced: `getL1FeeUpperBound(uint256)`. This method returns an upper bound for the L1 fee
+for a given transaction size. It is provided for callers who wish to estimate L1 transaction costs in the
+write path, and is much more gas efficient than `getL1Fee`. It assumes the worst case of no compression,
+implemented as follows:
+
+```solidity
+function getL1FeeUpperBound(uint256 unsignedTxSize) external view returns (uint256) {
```

Comment:
this may also be helpful for protocols that currently depend on constants directly and are doing their own L1 fee calculation (example: https://sips.synthetix.io/sips/sip-356/)